client/src/components/layout/UserProfile.js

```diff
@@ -26,12 +26,58 @@ const UserProfile = (props) => {
     getUserData();
   }, []);
 
+  const deleteReview = async (reviewId) => {
+    try {
+      const response = await fetch(`/api/v1/users/${userId}/${reviewId}`, {
+        method: "DELETE",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+      if (!response.ok) {
+        const errorMessage = `${response.status} ${response.statusText}`;
+        const error = new Error(errorMessage);
+        throw error;
+      }
+      getUserData();
```

Comment:
this means you have to make ANOTHER request, despite your DELETE request having access to everything you need on the backend.

I see why you have `getUserData` here, but its costly. Instead, in the response from your delete request send back the same information you need from getUserData. That way, its one fewer request. so your delete endpoint is going to get busier with some user info/dogs/reviews.